addons/api/addon/serializers/auth-method.js

```diff
@@ -4,7 +4,28 @@
  */
 
 import ApplicationSerializer from './application';
-import { TYPE_AUTH_METHOD_OIDC } from 'api/models/auth-method';
+import { TYPE_AUTH_METHOD_PASSWORD } from 'api/models/auth-method';
+
+const fieldsByType = {
```

Comment:
I'm not the biggest fan of this since we can start utilizing the `for` field we've conceptualized already in the model and can generalize now. It also helps documentation for field types in the model rather than the serializer. 

Doing it this way requires us to keep adding similar logic to `serializeAttribute` for each serializer that needs this logic

Reply:
@ZedLi good point,  I like the concept of using `for` and was now of the reason asked that question in an earlier thread.  I will put in a change, and also add a ticket to switch over the `host-catalogs` serializer as well